python/src/liberate.py

```diff
@@ -95,6 +95,21 @@ def run_liberate(
     )
     results.check_returncode()
 
-    return LiberateResult(
+    return LiberateCompletedProcess(
         args=results.args, returncode=results.returncode, stdout=results.stdout
     )
+
+
+def liberate_simulator(candidates: Sequence[Netlist]) -> LibertyResult:
+    # Setup inputs
+    cell_names = tuple(netlist.cell_name for netlist in candidates)
+    template_tcl_file = File(TEMPLATE_TCL_DEFAULT_PATH)
+    update_liberate_template_cell_names(template_tcl_file, cell_names)
+
+    # Run simulations
+    run_liberate()  # TODO: paramaterize other args
```

Comment:
This TODO makes me think simulator should be a class with a `simulate` function so you can initialize the configuration when you create the class. Sure we can also accomplish this with `functools.partial`, but that essentially does the same thing (creates a partial object that holds some data and then injects it into the function when you call it) but less explicitly

Reply:
I think you wrote that TODO, this code is extracted from your simulate method from the param sweep. We wouldn't need functools.partial if we did want to parameterize, we could just use python default arguments.

Reply:
no need for a class :)

Reply:
I mean yeah we can hardcode for now. But if we do want to parametrize the liberate paths then something will need to be done, either `partial` or `class`. Fine leaving it for now though

Reply:
I see what you mean. I think you can guess how I feel about partial vs class hahaha. I will leave it for now